src/foam/dao/MDAO.java

```diff
@@ -107,11 +107,44 @@ public MDAO(ClassInfo of) {
     index_ = new AltIndex(new TreeIndex((PropertyInfo) this.of_.getAxiomByName("id")));
   }
 
+
+
+  public ArrayList<PropertyInfo> propertyInfoList_(Index index){
+        ArrayList<PropertyInfo> infoList = new ArrayList<PropertyInfo>();
+        var current = index;
+        while ( current != null && ( current instanceof TreeIndex ) ){
+          infoList.add(((TreeIndex)current).getProp());
+          current = ((TreeIndex)current).getTail();
+        }
+        return infoList;
+
+  }
+  public boolean isSubSet_(ArrayList<PropertyInfo> list1 , ArrayList<PropertyInfo> list2){
```

Comment:
This doesn't really have anything to do with an MDAO. It might be worth changing the method to 
```java
public static <T> boolean isSubSet(List<T> l1, List<T> l2) {
  if ( l1 == null || l2 == null ) return false;
  return l1.containsAll(l2);
}
```
and adding it to SafetyUtil